src/encoded/static/components/browse/components/file-tables.js

```diff
@@ -483,43 +485,55 @@ export class RawFilesStackedTable extends React.PureComponent {
             );
         }
 
-        var experimentVisibleName = (
-                exp.tec_rep_no ? 'Tech Replicate ' + exp.tec_rep_no :
-                    (exp.experiment_type && exp.experiment_type.display_title) ? exp.experiment_type.display_title :
-                        exp.accession
-            ),
-            experimentAtId  = object.itemUtil.atId(exp),
-            linkTitle       = !experimentAtId && exp.error ? <em>{ exp.error }</em> : experimentVisibleName,
-            // Shown in show more button, e.g. "Show X more Files".
-            listTitle       = (
-                (haveUngroupedFiles? "Files" : '') +
-                (haveUngroupedFiles && haveGroups? " or " : '') +
-                (haveGroups? "Groups" : '')
-            );
+        const experimentVisibleName = (
+            exp.tec_rep_no ? 'Tech Replicate ' + exp.tec_rep_no :
+                (exp.experiment_type && exp.experiment_type.display_title) ? exp.experiment_type.display_title :
+                    exp.accession
+        );
+        const experimentAtId  = object.itemUtil.atId(exp);
+        const linkTitle       = !experimentAtId && exp.error ? <em>{ exp.error }</em> : experimentVisibleName;
+        // Shown in show more button, e.g. "Show X more Files".
+        const listTitle       = (
+            (haveUngroupedFiles? "Files" : '') +
+            (haveUngroupedFiles && haveGroups? " or " : '') +
+            (haveGroups? "Groups" : '')
+        );
+
+
+        const showMoreExtTitle = preventExpand ? (
+            <a href={object.itemUtil.atId(experimentSet)}>(view in Experiment Set)</a>
+        ) : null;
 
         return (
             <StackedBlock key={ experimentAtId || exp.tec_rep_no || i } hideNameOnHover={false} columnClass="experiment" stripe={this.cache.oddExpRow}
                 label={<StackedBlockNameLabel title="Experiment" accession={exp.accession} subtitleVisible />}>
                 <StackedBlockName relativePosition={expFxn.fileCountFromSingleExperiment(exp) > 6}>
                     { experimentAtId ? <a href={experimentAtId} className="name-title">{ linkTitle }</a> : <span className="name-title">{ linkTitle }</span> }
                 </StackedBlockName>
-                <StackedBlockList title={listTitle} className={contentsClassName}
-                    collapseLimit={collapseLimit || 3} collapseShow={collapseShow || 2} collapseLongLists={collapseLongLists}>
+                <StackedBlockList {...{ collapseLimit, collapseShow, collapseLongLists, showMoreExtTitle }} title={listTitle} className={contentsClassName}>
                     { contents }
                 </StackedBlockList>
             </StackedBlock>
         );
     }
 
     renderBiosampleStackedBlockOfExperiments(expsWithBiosample,i){
-        const { collapseLimit, collapseShow } = this.props;
+        const { collapseLimit = 3, collapseShow = 2, experimentSet, preventExpand = false } = this.props;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['collapseShow' is missing in props validation](https://app.codacy.com/manual/4dn/fourfront/pullRequest?prid=4660599)